.github/workflows/build.yml

```diff
@@ -1,28 +1,314 @@
-# This workflow uses actions that are not certified by GitHub.
-# They are provided by a third-party and are governed by
-# separate terms of service, privacy policy, and support
-# documentation.
-# This workflow will download a prebuilt Ruby version, install dependencies and run tests with Rake
-# For more information see: https://github.com/marketplace/actions/setup-ruby-jruby-and-truffleruby
-
 name: Build
 
 on:
+  workflow_dispatch:
   push:
-    branches: [ $default-branch ]
-  pull_request:
-    branches: [ $default-branch ]
+
+env:
+  DISABLE_KNAPSACK: true
 
 jobs:
-  test:
+  test-non-spree-models:
```

Comment:
Can we use a matrix to DRY this? https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix

Reply:
that's meant for environment-specific stuff but we could give it a try. However, I'd prefer to do it separately so we can start getting feedback already while working on DRYing that.

Reply:
Another approach I soon discarded is having a file (and so a workflow) per job like https://github.com/decidim/decidim/tree/develop/.github/workflows. I'm not sure what's the best approach.